plugins/net_plugin/net_plugin.cpp

```diff
@@ -872,6 +887,13 @@ namespace eosio {
       block_id_type null_id;
       for (auto bid = head_id; bid != null_id && bid != lib_id; ) {
          try {
+
+            // if the last handshake received indicates that we are catching up on a fork
+            // that the peer is already partially aware of, no need to resend blocks
+            if (block_header::num_from_id(bid) == remote_head_num && remote_head_id == bid) {
```

Comment:
Is there a reason this is not just:
`if( remote_head_id == bid ) {`

Reply:
I suppose not given the recent updates to sha256 equality comparison, its probably more efficient to just do the `==`